nixos/modules/services/security/munge.nix

```diff
@@ -45,19 +45,25 @@ in
 
     systemd.services.munged = {
       wantedBy = [ "multi-user.target" ];
-      after = [ "network.target" ];
+      wants = [
+        "network-online.target"
+        "time-sync.target"
+      ];
+      after = [
+        "network-online.target"
+        "time-sync.target"
```

Comment:
The motivation is that `nixosTests.slurm` shows warnings such as `control # [   17.073884] slurmctld[769]: error: High latency for 1000 calls to gettimeofday(): 4676 microseconds`, and I was thinking we could add the optional dependency on `time-sync.target` to all of the slurm-related services, and we could also enable ntp in the nixos test. That'd clean up the test's logs, and the test could also serve as a piece of documentation for the nixos module

Reply:
> and we could also enable ntp in the nixos test

I think for the test itself the timing is not important.  I have never observed any problems with that. 

> thinking we could add the optional dependency on time-sync.target to all of the slurm-related services

In general: slurm does not seem to be sensitive to the system clock (at least not on startup).

Reply:
Synchronizing the time does remove the warnings, and logs without warnings are more convenient to read 🤷🏻

Reply:
synchronized time is actually important for munge. slurm doesn't work if system times differ too much.
I just checked our CentOS/Rocky cluster, and the systemd units for munge also wait for time-sync.target.
I have no preference wrt network-online vs network. For some units it's import to wait for network-online, but I haven't seen problems with munge here.

Reply:
I did observe `gettimeofday` marked as an "error" instead of a "warning" in the logs a few times, but only in conjunction with other failures, so it was hard to say which one was the primal cause. Ultimately, I'd like to just remove this one variable out of the equation